ballerina/commons.bal

```diff
@@ -261,56 +279,71 @@ public type SubscriptionTemporaryRedirect record {
 };
 
 # Record to represent the unsubscription acceptance.
+# 
+# + statusCode - HTTP status code for the response
 public type UnsubscriptionAccepted record {
     *CommonResponse;
+    int statusCode = http:STATUS_ACCEPTED;
 };
 
 # Record to represent the acknowledgement of content updated by the publisher.
+# 
+# + statusCode - HTTP status code for the response
 public type Acknowledgement record {
     *CommonResponse;
+    int statusCode = http:STATUS_OK;
 };
 
 # Common response, which could be used for `websubhub:TopicRegistrationSuccess`.
 public final readonly & TopicRegistrationSuccess TOPIC_REGISTRATION_SUCCESS = {};
 
 # Common response, which could be used for `websubhub:TopicRegistrationError`.
-public final TopicRegistrationError TOPIC_REGISTRATION_ERROR = error TopicRegistrationError("Topic registration failed");
+public final TopicRegistrationError TOPIC_REGISTRATION_ERROR = error TopicRegistrationError(
+    "Topic registration failed", statusCode = http:STATUS_OK);
 
 # Common response, which could be used for `websubhub:TopicDeregistrationSuccess`.
 public final readonly & TopicDeregistrationSuccess TOPIC_DEREGISTRATION_SUCCESS = {};
 
 # Common response, which could be used for `websubhub:TopicDeregistrationError`.
-public final TopicDeregistrationError TOPIC_DEREGISTRATION_ERROR = error TopicDeregistrationError("Topic deregistration failed!");
+public final TopicDeregistrationError TOPIC_DEREGISTRATION_ERROR = error TopicDeregistrationError(
+    "Topic deregistration failed!", statusCode = http:STATUS_OK);
 
 # Common response, which could be used for `websubhub:Acknowledgement`.
 public final readonly & Acknowledgement ACKNOWLEDGEMENT = {};
 
 # Common response, which could be used for `websubhub:UpdateMessageError`.
-public final UpdateMessageError UPDATE_MESSAGE_ERROR = error UpdateMessageError("Error in accessing content");
+public final UpdateMessageError UPDATE_MESSAGE_ERROR = error UpdateMessageError(
+    "Error in accessing content", statusCode = http:STATUS_OK);
 
 # Common response, which could be used for `websubhub:SubscriptionAccepted`.
 public final readonly & SubscriptionAccepted SUBSCRIPTION_ACCEPTED = {};
 
 # Common response, which could be used for `websubhub:BadSubscriptionError`.
-public final BadSubscriptionError BAD_SUBSCRIPTION_ERROR = error BadSubscriptionError("Bad subscription request");
+public final BadSubscriptionError BAD_SUBSCRIPTION_ERROR = error BadSubscriptionError(
```

Comment:
Shall we default the status codes for these errors? I feel like the status code is a bit redundant. If used does not need the default status code, they can override. We can created a specific Error detail for the `BadSubscriptionError` and do following
```ballerina
public type BadSubscriptionErrorDetail record {
    *CommonResponse
    int statusCode = http:STATUS_BAD_REQUEST;
};

public type BadSubscriptionError distinct (Error & error<BadSubscriptionErrorDetail>);
```

Reply:
@shafreenAnfar WDYT ?

Reply:
What if they want to change the error message? :) 
In that case they cant use websubhub:BAD_SUBSCRIPTION_ERROR right. However websubhub:BAD_SUBSCRIPTION_ERROR declaration also does not need the status code if you change the error definition as suggested

Reply:
My suggestion is to keep the defaults at the definition level. So both theses error constants and the user created one will have the defaults. With your approach, only constants have defaults but not the definition.

Reply:
Understood. Seems a good improvement to be added :thumbsup:

Reply:
It seems like there is a lang-level limitation for having intersection types with default values [1]

[1] - [https://github.com/ballerina-platform/ballerina-spec/issues/544#issuecomment-778099662](https://github.com/ballerina-platform/ballerina-spec/issues/544#issuecomment-778099662)